### DISCUSSION TITLE
Is there any way to add error handling to stream responses?

### DISCUSSION THREAD

**Discussion by danechitoaie** [litestar-org/litestar#1573] [Q&A]
Looking at https://docs.litestar.dev/1/usage/responses.html#streaming-responses

```python
from asyncio import sleep
from datetime import datetime
from collections.abc import AsyncGenerator

from starlite import Starlite, Stream, get
from starlite.utils.serialization import encode_json


async def my_generator() -> AsyncGenerator[bytes, None]:
    while True:
        await sleep(0.01)
        yield encode_json({"current_time": datetime.now()})


@get(path="/time")
def stream_time() -> Stream:
    return Stream(iterator=my_generator())


app = Starlite(route_handlers=[stream_time])
```

What if `encode_json({"current_time": datetime.now()})` raises an exception? It seems the status code is always 201.

**Comment by Goldziher** [comment]
Have you encountered an actual error with this? Please add a PR with a reproduction

**Reply by danechitoaie** [reply]
I was not referring to the exact above example. I was asking in general. Let's say we have the following code.
```python
from asyncio import sleep
from datetime import datetime
from collections.abc import AsyncGenerator

from starlite import Starlite, Stream, get
from starlite.utils.serialization import encode_json


async def my_generator() -> AsyncGenerator[bytes, None]:
    i = 0
    while True:
        await sleep(0.01)
        i += 1
        if i == 3:
            raise Exception("Error!")
        yield encode_json({"current_time": datetime.now()})


@get(path="/time")
def stream_time() -> Stream:
    return Stream(iterator=my_generator())


app = Starlite(route_handlers=[stream_time])
```

What happens when the exception is raised? How can the client handle this error? As the stream starts, so you get the first 2 JSON outputs, but on 3rd one you get an exception and response is closed and on client side you still remain with the 201 status code that request was ok.

But I think is is more of a HTTP protocol issue when the output is in chuck mode. I don't think you can send another status code for the request since you already sent one when you start the stream.
